tests/unit/test_snap.py

```diff
@@ -661,3 +661,40 @@ def test_install_local_args(self, mock_subprocess):
                 universal_newlines=True,
             )
             mock_subprocess.reset_mock()
+
+    @patch("charms.operator_libs_linux.v2.snap.subprocess.check_output")
+    def test_alias(self, mock_subprocess):
+        mock_subprocess.return_value = ""
+        foo = snap.Snap("foo", snap.SnapState.Latest, "stable", "1", "classic")
+        foo.alias("bar", "baz")
+        mock_subprocess.assert_called_once_with(
+            ["snap", "alias", "foo.bar", "baz"],
+            universal_newlines=True,
+        )
+        mock_subprocess.reset_mock()
+
+        foo.alias("bar")
+        mock_subprocess.assert_called_once_with(
+            ["snap", "alias", "foo.bar", "bar"],
+            universal_newlines=True,
+        )
+        mock_subprocess.reset_mock()
+
+    @patch("charms.operator_libs_linux.v2.snap.subprocess.check_output")
+    def test_alias_raises_snap_error(self, mock_subprocess):
+        mock_subprocess.side_effect = CalledProcessError(
+            returncode=1, cmd=["snap", "alias", "foo.bar", "baz"]
+        )
+        foo = snap.Snap("foo", snap.SnapState.Latest, "stable", "1", "classic")
+        with self.assertRaises(snap.SnapError) as ctx:
+            foo.alias("bar", "baz")
+        self.assertEqual("<charms.operator_libs_linux.v2.snap.SnapError>", ctx.exception.name)
+        self.assertEqual(
```

Comment:
I think testing the specifics of the error string is too much. It's not part of the API and users shouldn't be relying on it. I'd also probably drop the exception name test. We're already testing that it's an instance of `SnapError` -- that's the API contract, and seems to me all we should be testing here.

Reply:
Changed to only checking that the correct exception was raised.